models/articles.js

```diff
@@ -24,27 +24,28 @@ exports.fetchArticles = (topic,sort_by = 'created_at',order = 'desc') => {
 }
 
 exports.fetchArticleById = (article_id) => {
-    
     if(!article_id.match(/^[0-9]*$/gm)){
         return Promise.reject({
             status: 400,
             msg: 'invalid request for article id'
         })
     }
-    return db.query(
-        `
-        SELECT * FROM articles
-        WHERE article_id = $1;
-        `,[article_id]
-    ).then((result) => {
-        if(result.rows.length === 0){
-            return Promise.reject({
-                status: 404,
-                msg: 'invalid request for article id'
-            })
-        }
-        return result.rows[0];
-    });
+    return checkArticleExists(article_id)
+        .then(() =>{
+        return addCommentsCountToArticle(article_id).then((res) => {
```

Comment:
We don't need to actually add the comments_count to the article table as we can get hold of it by doing a COUNT, JOIN and GROUP BY and in the query. We should leave the tables unaltered.